project/webapp/controller/ProductsList.controller.js

```diff
@@ -220,19 +220,35 @@ sap.ui.define(
           oTableBinding.filter(null);
         },
 
-        _deleteProduct() {
-          // TODO: add products delete functionality
-        },
-
-        onDeleteProductPress() {
-          MessageBox.confirm(this._getConfirmationText(), {
-            title: this.oResourceBundle.getText("ConfirmDeleteProductTitle"),
-            actions: [MessageBox.Action.OK, MessageBox.Action.CLOSE],
-            onClose: (sAction) => {
-              switch (sAction) {
-                case MessageBox.Action.OK:
-                  this._deleteProduct();
-                  break;
+      _removeProductsFromList(aProductsList) {
+        const aSelectedProducts = this.byId("productList").getSelectedItems();
+        
+        aSelectedProducts.map((item) => {
+          const sProductId = item.getBindingContext(this.APP_MODEL_NAME).getProperty("id");
+          aProductsList = aProductsList.filter(({id}) => id !== sProductId);
```

Comment:
that's a bad practice to reassign function parameter, create new variable instead (aNewProductList for example)